osquery/logger/plugins/tests/buffered_tests.cpp

```diff
@@ -271,4 +276,74 @@ TEST_F(BufferedLogForwarderTests, test_split) {
       .WillOnce(Return(Status(0)));
   runner2.check();
 }
+
+// Test the purge() function independently of check()
+TEST_F(BufferedLogForwarderTests, test_purge) {
+  FLAGS_buffered_log_max = 3;
+  StrictMock<MockBufferedLogForwarder> runner("mock", kLogPeriod, 100);
+  for (uint64_t i = 0; i < 10; ++i) {
+    runner.logString(std::to_string(i));
+    StatusLogLine log1 = makeStatusLogLine(O_INFO, "foo", 1, "foo status");
+    StatusLogLine log2 = makeStatusLogLine(O_ERROR, "bar", 30, "bar error");
+    runner.logStatus({log1, log2});
+  }
+  // wait for a new timestamp
+  std::this_thread::sleep_for(std::chrono::milliseconds(1500));
```

Comment:
We could do this, but it would involve exposing an extra parameter in the `logStatus`, `logString` and `genIndex` functions, or else making `genIndex` virtual so we can override it in the mock class. I will do one of those if you think it's worth it.